config.go

```diff
@@ -82,61 +69,68 @@ func (c *Config) SetLabels(labels ...string) {
 	c.Labels = labels
 }
 
-func (c Config) Clone() Config {
+func (c *Config) Clone() Config {
+	c.ensureLifecycleListeners()
+	c.ensureMembershipListeners()
 	newLabels := make([]string, len(c.Labels))
 	copy(newLabels, c.Labels)
 	return Config{
-		ClientName:          c.ClientName,
-		Labels:              newLabels,
-		ClusterConfig:       c.ClusterConfig.Clone(),
-		SerializationConfig: c.SerializationConfig.Clone(),
-		LoggerConfig:        c.LoggerConfig.Clone(),
-		StatsConfig:         c.StatsConfig.clone(),
+		ClientName:    c.ClientName,
+		Labels:        newLabels,
+		Cluster:       c.Cluster.Clone(),
+		Serialization: c.Serialization.Clone(),
+		Logger:        c.Logger.Clone(),
+		Stats:         c.Stats.clone(),
 		// both lifecycleListeners and membershipListeners are not used verbatim in client creator
 		// so no need to copy them
 		lifecycleListeners:  c.lifecycleListeners,
 		membershipListeners: c.membershipListeners,
 	}
 }
 
-func (c Config) Validate() error {
-	if err := c.ClusterConfig.Validate(); err != nil {
+func (c *Config) Validate() error {
+	if err := c.Cluster.Validate(); err != nil {
 		return err
 	}
-	if err := c.SerializationConfig.Validate(); err != nil {
+	if err := c.Serialization.Validate(); err != nil {
 		return err
 	}
-	if err := c.LoggerConfig.Validate(); err != nil {
+	if err := c.Logger.Validate(); err != nil {
 		return err
 	}
-	if err := c.StatsConfig.Validate(); err != nil {
+	if err := c.Stats.Validate(); err != nil {
 		return err
 	}
 	return nil
 }
 
+func (c *Config) ensureLifecycleListeners() {
+	if c.lifecycleListeners == nil {
+		c.lifecycleListeners = map[types.UUID]LifecycleStateChangeHandler{}
+	}
+}
+
+func (c *Config) ensureMembershipListeners() {
+	if c.membershipListeners == nil {
+		c.membershipListeners = map[types.UUID]cluster.MembershipStateChangeHandler{}
+	}
+}
+
 // StatsConfig contains configuration for Management Center.
 type StatsConfig struct {
 	// Enabled enables collecting statistics.
-	Enabled bool
+	Enabled bool `json:",omitempty"`
```

Comment:
I am not 100% sure if we should do the same here but in the Java client, the statistics are enabled by default and have the default period of 5 seconds

Reply:
It seems stats is off by default in the Python client 🤔

Reply:
Honestly, I put `Enabled` setting just to be compatible with other clients. I think it would be better to assume it is on if Period > 0

Reply:
I am OK with that